packages/block-library/src/comments-query-loop/edit.js

```diff
@@ -3,50 +3,71 @@
  */
 import {
 	BlockControls,
-	InspectorControls,
 	useBlockProps,
 	useInnerBlocksProps,
+	store as blockEditorStore,
 } from '@wordpress/block-editor';
-import { SelectControl } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
-import QueryToolbar from './toolbar';
+import CommentsToolbar from './edit/comments-toolbar';
+import CommentsInspectorControls from './edit/comments-inspector-controls';
+import { useSelect, useDispatch } from '@wordpress/data';
 
 const TEMPLATE = [
 	[ 'core/comment-template' ],
 	[ 'core/comments-pagination' ],
 ];
 
 export default function CommentsQueryLoopEdit( { attributes, setAttributes } ) {
-	const { perPage, tagName: TagName } = attributes;
+	const { tagName: TagName, inherit, perPage, order } = attributes;
 
 	const blockProps = useBlockProps();
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		template: TEMPLATE,
 	} );
 
+	const { __unstableMarkNextChangeAsNotPersistent } = useDispatch(
+		blockEditorStore
+	);
+
+	const {
+		__experimentalDiscussionSettings: { commentOrder, commentsPerPage },
+	} = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		return getSettings();
+	} );
+
+	useEffect( () => {
+		// We set the default value of order to the value of the discussion setting if it is not already defined.
+		if ( order === null && inherit ) {
+			__unstableMarkNextChangeAsNotPersistent();
+			setAttributes( { order: commentOrder } );
+		}
+		if ( perPage === null && inherit ) {
+			__unstableMarkNextChangeAsNotPersistent();
+			setAttributes( { perPage: commentsPerPage } );
+		}
+		// We set defaults when user switches to inherit int the block options.
+		if ( inherit ) {
+			setAttributes( { order: commentOrder, perPage: commentsPerPage } );
+		}
+	}, [ inherit ] );
```

Comment:
I'm not convinced we really need those side effects here. When the `inherit` value is set to `false` (default) then we can read values directly from settings in the components that render controls. In case the user toggles the `inherit` control then we can set the new value for `inherit`. In addition, we can use the same `setAttributes` call to  set values for `order` and `perPage` to `null`, when toggling to `true`, and to values read from settings, when toggling to `false`.